src/charm.py

```diff
@@ -81,22 +88,25 @@ def _on_config_changed(self, _: ConfigChangedEvent) -> None:  # noqa: C901
         for option in S3_OPTIONS:
             # experimental config options should be handled with the "experimental-" prefix
             if option == "delete-older-than-days" and f"experimental-{option}" in self.config:
-                config_value = str(self.config[f"experimental-{option}"])
-                # reset value if empty config_value
-                if config_value == "" and self.get_secret("app", option) is not None:
-                    self.set_secret("app", option, None)
-                    update_config.update({option: ""})
-                elif config_value != "":
-                    update_config.update({option: config_value})
-                    self.set_secret("app", option, config_value)
+                config_value = self.config[f"experimental-{option}"]
+                # check if new config value is inside allowed range
+                if config_value > 0 and config_value <= MAX_RETENTION_DAYS:
+                    update_config.update({option: str(config_value)})
+                    self.set_secret("app", option, str(config_value))
+                else:
+                    logger.warning(
+                        "Invalid value %s for config 'delete-older-than-days', ignoring.",
+                        config_value,
+                    )
```

Comment:
One last improvement: a `BlockedStatus` may be useful for the user to know that the config value is invalid (so they don't need to check the logs to notice that).